README.md

```diff
@@ -57,48 +60,253 @@ This is work in progress as of 14 Nov 2018:
+  })
+  getUser(@pathParam({ description: "Unique user identifier" }) id: string): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "POST",
+    path: "/users",
+    description: "Create a user",
+    tags: ["Users"]
+  })
+  @specificError<ApiError>({
+    name: "unauthorized",
+    statusCode: 401
+  })
+  createUser(
+    @request req: CreateUserRequest,
+    @header({
+      name: "Authorization",
+      description: "This is the authorization token"
+    })
+    authToken: Optional<string>
+  ): CreateUserResponse {
+    return response();
+  }
+}
+
+interface UserResponse {
+  firstName: string;
+  lastName: string;
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface CreateUserResponse {
+  success: boolean;
+}
+
+interface ApiError {
+  message: string;
+}
+```
+
+| Field                | Default            | Description                                               |
+| -------------------- | ------------------ | --------------------------------------------------------- |
+| `method`             |                    | (**required**) [HTTP method](#suppported-http-methods)    |
+| `path`               |                    | (**required**) URL path                                   |
+| `description`        | `""`               | Description of the endpoint                               |
+| `requestContentType` | `application/json` | Content type of the request body                          |
+| `successStatusCode`  | `200`              | HTTP status code for a successful response                |
+| `tags`               |                    | Array of tags used for endpoint grouping in documentation |
+
+### `@request`
+
+Define a request body for requests that require one. This is commonly used for `POST` and `PUT` requests:
+
+```TypeScript
+class MyUserEndpoints {
+  //...
+  @endpoint({
+    method: "POST",
+    path: "/users"
+  })
+  createUser(
+    @request req: CreateUserRequest
+  ): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "PUT",
+    path: "/users/:id"
+  })
+  updateUser(
+    @pathParam({ description: "User unique identifier" }) id: string,
+    @request req: UpdateUserRequest
+  ): UserResponse {
+    return response();
+  }
+  //...
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface UpdateUserRequest {
+  lastName: string;
+}
+```
+
+### `@header`
+
+Define a request header. `@header` can be used multiple times to define multiple headers.
+
+| Field         | Description                       |
+| ------------- | --------------------------------- |
+| `name`        | (**required**) Name of the header |
+| `description` | Description of the header         |
+
+### `@pathParam`
+
+Define path parameters that appear in the `@endpoint.path`. The parameter must be defined in the `@endpoint.path`. A `@pathParam` should be defined for each parameter defined in `@endpoint.path`.
```

Comment:
It may not be super obvious that `@endpoint.path` refers to `@endpoint({ path: ... })` so I'd suggest the following:
```suggestion
Define path parameters that appear in the `path` provided in `@endpoint()`. For example if the path is `/users/:userId`, the endpoint method must define a matching argument with `@pathParam() userId: string`.

Note: the name of the argument must match the name of the path parameter.
```